sdk/src/main/java/com/deploygate/sdk/SdkDeviceStatesCollector.java

```diff
@@ -43,10 +51,13 @@ public void collectLocale() {
 
     private void putState(String fqcn, String paramName, Object data) {
         String key = String.format("%s$%s", fqcn, paramName);
-        try {
-            states.put(key, data);
-        } catch (JSONException e) {
-            Logger.w(e, "Failed to put info: key=%s, value=%s", key, data);
+
+        synchronized (mLock) {
+            try {
+                states.put(key, data);
+            } catch (JSONException e) {
+                Logger.w(e, "Failed to put info: key=%s, value=%s", key, data);
+            }
         }
```

Comment:
synchronized block should be small as much as possible.

```suggestion
        try {
            synchronized (mLock) {
                states.put(key, data);
            }
        } catch (JSONException e) {
            Logger.w(e, "Failed to put info: key=%s, value=%s", key, data);
        }
```